src/main/java/io/github/jsonSnapshot/SnapshotDataItem.java

```diff
@@ -0,0 +1,67 @@
+package io.github.jsonSnapshot;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import lombok.NonNull;
+import lombok.Value;
+
+@Value
+public class SnapshotDataItem implements Comparable<SnapshotDataItem> {
+  private static final int REGEX_FLAGS = Pattern.MULTILINE + Pattern.DOTALL;
+  private static final Pattern REGEX =
+      Pattern.compile("(?<name>[^ =]*) *=+ *(?<data>\\[.*\\])[^\\]]*", REGEX_FLAGS);
+
+  public static SnapshotDataItem ofRawData(@NonNull final String rawDataString)
+      throws IllegalRawDataException {
+    final Matcher matcher = REGEX.matcher(rawDataString);
+    if (!matcher.matches()) {
+      throw new IllegalRawDataException(rawDataString);
+    }
+
+    final String name = matcher.group("name");
+    final String data = matcher.group("data");
+    return SnapshotDataItem.ofNameAndData(trimOrNull(name), trimOrNull(data));
+  }
+
+  private static String trimOrNull(final String str) {
+    if (str == null) {
+      return null;
+    }
+    return str.trim();
+  }
+
+  public static SnapshotDataItem ofNameAndData(
+      @NonNull final String name, @NonNull final String data) {
+    return new SnapshotDataItem(name, data);
+  }
+
+  String name;
+
+  String data;
+
+  private SnapshotDataItem(@NonNull final String name, @NonNull final String data) {
+    this.name = name;
+    this.data = data;
+  }
+
+  public String asRawData() {
+    return name + "=" + data;
+  }
+
+  @Override
+  public int compareTo(final SnapshotDataItem o) {
+    if (o == null) {
+      return 1;
+    }
+    return this.name.compareTo(o.name);
+  }
+
+  public static final class IllegalRawDataException extends IllegalArgumentException {
```

Comment:
Consider removing this exception subclass.

If this exception is unchecked, we don't expect the user to catch it.
If that is the case, then `IllegalArgumentException` is as informative as `IllegalRawDataException`.

Reply:
@grimsa 
hm..i think the class name `IllegalRawDataException` makes it more easy to understand what the string we create the exception with is about (e.g. it is not an illegal snapshot *name*).
Pls let me know what you think..if you still think i should change it to IllegalArgumentException (i don't think it should be a checked one), no problem doing that.

Reply:
If the exception is not expected to be catched ever, I'd do something like `throw new IllegalArgumentException("Raw data string does not match expected pattern. String: " + rawDataString);`

I'd create a dedicated class if I was handling it meaningfully somewhere.